bot/src/bot.js

```diff
@@ -1,99 +1,96 @@
 import { Client } from  'irc';
 
-const CONFIRMATION_MESSAGES = [
-  'naturally',
-  'of course',
-  'splendid',
-  'jolly good',
-  'noted',
-  'point taken',
-  'at your service',
-  'sounds good'
-];
+import { Commands } from './commands';
 
-const ERROR_MESSAGES = [
-  'uh-oh, that didn\'t work out',
-  'something went wrong'
-];
-
-function randElem(arr) {
-  return arr[Math.floor(Math.random() * arr.length)];
-}
-
-function saveUpdate(db, author, channel, text) {
-  const d = new Date();
-  return db.collection('updates').insert({
-    author,
-    channel,
-    status: 'inbox',
-    resolved: false,
-    text,
-    createdAt: d,
-    firstCreatedAt: d,
-  }).then(
-    () => randElem(CONFIRMATION_MESSAGES),
-    () => randElem(ERROR_MESSAGES)
-  );
+function testCommand(bot, author, channel, message) {
+  for (let command of bot.commands) {
+    if (command.matches(message)) {
+      return Promise.resolve(command.test(bot, author, channel, message));
+    }
+  }
+  return Promise.resolve(null);
 }
 
-const createDateRe = /create a report for ([0-9]{4}-[0-9]{2}-[0-9]{2})/;
-
-function createReport(db, text) {
-  const [, slug] = createDateRe.exec(text);
-  const ts = Date.parse(slug);
-
-  if (isNaN(ts)) {
-    return Promise.resolve('that\'s not a valid date');
+function parseCommand(bot, author, channel, message) {
+  console.log(message);
```

Comment:
Would you mind changing this to something like:

    console.log(` --- got a command: ${message}`);